docker-compose.yml

```diff
@@ -160,3 +160,18 @@ services:
     depends_on:
       - handshake
       - handshake-api
+
+  mongo:
+    image: mongo:4.4.1
+    command: --keyFile=/data/mgkey --replSet=skynet
+    container_name: mongo
+    restart: unless-stopped
+    logging: *default-logging
+    volumes:
+      - ./docker/data/mongo/db:/data/db
+      - ./docker/data/mongo/mgkey:/data/mgkey:rw
+    networks:
+      shared:
+        ipv4_address: 10.10.10.70
+    ports:
+      - "27017:27017"
```

Comment:
we should expose the ports via https, easiest thing to do is to proxy it through caddy

Reply:
Mongo typically uses its own SSL setup with those X.509 certificates. I don't think we should be circumventing that because the current setup is using the equivalent of a password file on disk, which is not optimal.

On the other hand the password in that file is 1024bit, so it's fairly secure. But the recommended production setup is definitely X.509-based. See https://docs.mongodb.com/manual/core/security-x.509/

Reply:
I don't think that works. Our caddy certificate is just a server cert from let's encrypt. MongoDB uses the certs not just for encryption but also for access control. So we should probably follow up with a self signed root certificate that we add to our instances and individual client certs for each portal which is signed by the private key of that root certificate.

Reply:
To make it a bit harder to sniff our open port we could also change it to a different port than the default one and set secure IPs from which we allow connections.

Reply:
The second part is already done manually - the mongo port is only accessible from our other portal nodes. Or at least so I thought before learning that Docker can add its own exceptions to the firewall rules.

Let's start with this setup and I'll work on adding the certificates in a F/U.

Reply:
The F/U ticket is #515